userspace/libsinsp/test/container_engine/container_parser_cri_containerd.ut.cpp

```diff
@@ -0,0 +1,911 @@
+	(*labels)["io.kubernetes.pod.uid"] = "hdishddjaidwnduw9a43535366368";
+	(*labels)["io.kubernetes.pod.namespace"] = "default";
+	(*labels)["io.kubernetes.pod.name"] = "nginx-sandbox";
+	auto annotations = status->mutable_annotations();
+	(*annotations)["io.kubernetes.container.restartCount"] = "0";
+	(*annotations)["io.kubernetes.container.hash"] = "4689db3";
+	(*annotations)["io.kubernetes.pod.terminationGracePeriod"] = "30";
+	status->mutable_metadata()->set_name("busybox");
+	runtime::v1alpha2::Mount mount;
+	mount.set_container_path("/host/boot");
+	mount.set_host_path("/boot");
+	mount.set_readonly(true);
+	mount.set_selinux_relabel(false);
+	mount.set_propagation(runtime::v1alpha2::MountPropagation::PROPAGATION_PRIVATE);
+	status->mutable_mounts()->Add()->CopyFrom(mount);
+	resp.mutable_info()->insert({"info", container_info_json});
+
+	return resp;
+}
+
+runtime::v1alpha2::PodSandboxStatusResponse get_default_cri_containerd_pod_status_resp()
+{
+
+	//     "status": {
+	//     "id": "63060edc2d3aa803ab559f2393776b151f99fc5b05035b21db66b3b62246ad6a",
+	//     "metadata": {
+	//       "attempt": 1,
+	//       "name": "nginx-sandbox",
+	//       "namespace": "default",
+	//       "uid": "hdishddjaidwnduw9a43535366368"
+	//     },
+	//     "state": "SANDBOX_READY",
+	//     "createdAt": "2023-12-12T00:15:14.434884181Z",
+	//     "network": {
+	//       "additionalIps": [],
+	//       "ip": "10.244.0.2"
+	//     },
+	//     "linux": {
+	//       "namespaces": {
+	//         "options": {
+	//           "ipc": "POD",
+	//           "network": "POD",
+	//           "pid": "POD",
+	//           "targetId": ""
+	//         }
+	//       }
+	//     },
+	//     "labels": {},
+	//     "annotations": {},
+	//     "runtimeHandler": ""
+	//   },
+
+	// Mock container runtime socket API responses PodSandboxStatusResponse
+	// Note that some fields are not populated or tested
+	runtime::v1alpha2::PodSandboxStatusResponse resp;
+
+	auto status = resp.mutable_status();
+	status->set_id("63060edc2d3aa803ab559f2393776b151f99fc5b05035b21db66b3b62246ad6a");
+	status->set_state(runtime::v1alpha2::PodSandboxState::SANDBOX_READY);
+	status->mutable_metadata()->set_name("nginx-sandbox");
+	resp.mutable_info()->insert({"info", pod_info_json});
+	status->mutable_network()->set_ip("10.244.0.2");
+	auto labels = status->mutable_labels();
+	(*labels)["io.kubernetes.pod.uid"] = "hdishddjaidwnduw9a43535366368";
+	(*labels)["io.kubernetes.pod.namespace"] = "default";
+	(*labels)["io.kubernetes.pod.name"] = "nginx-sandbox";
+	auto metadata = status->mutable_metadata();
+	metadata->set_attempt(0);
+	metadata->set_name("nginx-sandbox");
+	metadata->set_namespace_("default");
+	metadata->set_uid("hdishddjaidwnduw9a43535366368");
+
+	return resp;
+}
+
+TEST_F(sinsp_with_test_input, container_parser_cri_containerd)
+{
+	std::string cri_path = "/run/containerd/containerd_mock.sock";
+	auto cri_api_v1alpha2 = std::make_unique<libsinsp::cri::cri_interface_v1alpha2>(cri_path);
```

Comment:
@gnosek I see to get some CI failures, but it worked locally on my box. I see you last refactored this code. Would have a suggestion what could be the cause for failures like below? Thanks in advance!

```
Linking CXX executable unit-test-libsinsp
/usr/bin/ld: CMakeFiles/unit-test-libsinsp.dir/container_engine/container_parser_cri_containerd.ut.cpp.o: in function `sinsp_with_test_input_container_parser_cri_containerd_Test::TestBody()':
/__w/libs/libs/userspace/libsinsp/test/container_engine/container_parser_cri_containerd.ut.cpp:792: undefined reference to `libsinsp::cri::cri_interface<libsinsp::cri::cri_api_v1alpha2>::parse_cri_mounts(runtime::v1alpha2::ContainerStatus const&, sinsp_container_info&)'
```

Reply:
This should be linked via libsinsp/container_engine/cri.cpp. If you don't link the whole container engine, you can probably just `#include <container_engine/cri.hpp>` in your unit test.

Reply:
:+1: had to convert all functions to inline, the include guard alone did not work for the .hpp file.